### PR TITLE
[feature] support deepep

### PR DESCRIPTION
**Pull request by khalil2ji3mp6**
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/moe/ep_moe/layer.py [hunk 1]
```diff
@@ -1318,8 +1325,189 @@ def forward_deepgemm_masked(
...
+        reorder_topk_ids: torch.Tensor,
+        seg_indptr: torch.Tensor,
+        masked_m: torch.Tensor,
+        expected_m: int,
+        num_recv_tokens_per_expert: List[int],
+        forward_batch: ForwardBatch,
+    ):
+        assert self.quant_method is not None
+        assert self.activation == "silu"
+        output_dtype = torch.bfloat16
```
Comment (xiaobaicxy) [inline python/sglang/srt/layers/moe/ep_moe/layer.py:1463]: Are there any precision issues when the input is in fp16?

File: python/sglang/srt/distributed/parallel_state.py [hunk 2]
```diff
@@ -573,7 +573,7 @@ def _all_gather_into_tensor(self, output: torch.Tensor, input: torch.Tensor):
             )
 
     def all_gather_into_tensor(self, output: torch.Tensor, input: torch.Tensor):
-        if not supports_custom_op():
+        if is_npu() or not supports_custom_op():
```
Comment (Hexq0210) [inline python/sglang/srt/distributed/parallel_state.py:576]: explain why add this adjustment

File: python/sglang/srt/layers/moe/ep_moe/token_dispatcher.py [hunk 3]
```diff
@@ -546,6 +548,10 @@ def dispatch_b(
 
         reorder_topk_ids = seg_indptr = None
 
+        # generate seg_indptr need to be returned for dp/ep Ascend
+        if is_npu():
+            seg_indptr = self.handle[1]
+
         return (
```
Comment (Hexq0210) [inline python/sglang/srt/layers/moe/ep_moe/token_dispatcher.py:555]: It's best to abstract the entire handle.

File: python/sglang/srt/layers/moe/topk.py [hunk 4]
```diff
@@ -244,10 +244,11 @@ def forward_npu(
 
         # NOTE: now npu_moe_gating_top_k can only support `group_count=256` pattern
         if global_num_experts == 256:
+            router_logits = router_logits.to(torch.float32)
```
Comment (Hexq0210) [inline python/sglang/srt/layers/moe/topk.py:247]: Why Do I Add a Type Forced Transfer Operation?

File: python/sglang/srt/layers/quantization/w8a8_int8.py [hunk 5]
```diff
@@ -981,3 +984,45 @@ def apply(
             topk_ids=topk_ids,
             top_k=topk_ids.shape[1],
         )
+
+
+class NPU_W8A8EPMoEMethod(NPU_W8A8MoEMethod):
```
Comment (Hexq0210) [inline python/sglang/srt/layers/quantization/w8a8_int8.py:989]: Change the name.

File: python/sglang/srt/layers/attention/ascend_backend.py [hunk 6]
```diff
@@ -75,6 +75,9 @@ def init_forward_metadata(self, forward_batch: ForwardBatch):
             )
         self.forward_metadata.seq_lens_cpu_int = forward_batch.seq_lens_cpu.int()
```
Comment (Hexq0210) [inline python/sglang/srt/layers/attention/ascend_backend.py:76]: delete AscendDeepEPDispatcher
